src/PowerShellEditorServices/Services/PowerShellContext/Handlers/GetCommandHandler.cs

```diff
@@ -36,41 +40,113 @@ internal class GetCommandHandler : IGetCommandHandler
+            // Get-Command -CommandType Function,Cmdlet,ExternalScript | Sort-Object -Property Name
             psCommand
                 .AddCommand("Microsoft.PowerShell.Core\\Get-Command")
                     .AddParameter("CommandType", new[] { "Function", "Cmdlet", "ExternalScript" })
-                .AddCommand("Microsoft.PowerShell.Utility\\Select-Object")
-                    .AddParameter("Property", new[] { "Name", "ModuleName" })
                 .AddCommand("Microsoft.PowerShell.Utility\\Sort-Object")
                     .AddParameter("Property", "Name");
 
-            IEnumerable<PSObject> result = await _powerShellContextService.ExecuteCommandAsync<PSObject>(psCommand).ConfigureAwait(false);
+            IEnumerable<CommandInfo> result = await _powerShellContextService.ExecuteCommandAsync<CommandInfo>(psCommand).ConfigureAwait(false);
 
             var commandList = new List<PSCommandMessage>();
             if (result != null)
             {
-                foreach (dynamic command in result)
+                foreach (CommandInfo command in result)
                 {
+                    if (!_skipPackageManagementCheck && command.ModuleName == "PackageManagement" && command.Version < new Version(1, 4, 6))
+                    {
+                        _logger.LogDebug("Old version of PackageManagement detected. Attempting to update.");
+
+                        var takeActionText = "Yes";
+                        MessageActionItem messageAction = await _languageServer.Window.ShowMessage(new ShowMessageRequestParams
+                        {
+                            Message = "You have a version of PackageManagement that causes issues with the PowerShell extension. Would you like to update PackageManagement (You will need to restart the PowerShell extension after)?",
+                            Type = MessageType.Warning,
+                            Actions = new []
+                            {
+                                new MessageActionItem
+                                {
+                                    Title = takeActionText
+                                },
+                                new MessageActionItem
+                                {
+                                    Title = "Not now"
+                                }
+                            }
+                        });
+
+                        // If the user chose "Not now" ignore it for the rest of the session.
+                        if (messageAction?.Title != takeActionText)
+                        {
+                            _skipPackageManagementCheck = true;
+                        }
+                        else
+                        {
+                            // Update PackageManagement
+                            psCommand
+                                .AddCommand("PowerShellGet\\Install-Module")
```

Comment:
In Windows PowerShell I suspect without `-Scope CurrentUser` this will not work for most users (since it will require Admin)